cms/envs/common.py

```diff
@@ -326,6 +326,10 @@
 
     # Instead of AuthenticationMiddleware, we use a cache-backed version
     'cache_toolbox.middleware.CacheBackedAuthenticationMiddleware',
+    # Enable SessionAuthenticationMiddleware in order to invalidate
+    # user sessions after a password change.
+    'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
```

Comment:
Nevermind, it seems this is actually the recommended way: https://docs.djangoproject.com/en/1.7/topics/auth/default/#session-invalidation-on-password-change

I should have also read my earlier comment in the ticket.